server/app/filters/SessionTimeoutFilter.java

```diff
@@ -0,0 +1,141 @@
+package filters;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import auth.CiviFormProfile;
+import auth.ProfileUtils;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.nio.charset.StandardCharsets;
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Base64;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+import javax.inject.Inject;
+import javax.inject.Provider;
+import org.apache.pekko.stream.Materializer;
+import play.libs.Json;
+import play.mvc.Filter;
+import play.mvc.Http;
+import play.mvc.Result;
+import services.settings.SettingsManifest;
+
+public class SessionTimeoutFilter extends Filter {
+  public static final int DEFAULT_INACTIVITY_TIMEOUT_MINUTES = 30;
+  public static final int DEFAULT_MAX_SESSION_DURATION_MINUTES = 600;
+
+  private static final String TIMEOUT_COOKIE_NAME = "session_timeout_data";
+  private static final Duration COOKIE_MAX_AGE = Duration.ofDays(2);
+
+  private final ProfileUtils profileUtils;
+  private final Provider<SettingsManifest> settingsManifest;
+  private final Clock clock;
+
+  @Inject
+  public SessionTimeoutFilter(
+      Materializer mat,
+      ProfileUtils profileUtils,
+      Provider<SettingsManifest> settingsManifest,
+      Clock clock) {
+    super(mat);
+    this.profileUtils = checkNotNull(profileUtils);
+    this.settingsManifest = checkNotNull(settingsManifest);
+    this.clock = checkNotNull(clock);
+  }
+
+  @Override
+  public CompletionStage<Result> apply(
+      Function<Http.RequestHeader, CompletionStage<Result>> nextFilter,
+      Http.RequestHeader requestHeader) {
+
+    Optional<CiviFormProfile> profileOpt = profileUtils.optionalCurrentUserProfile(requestHeader);
```

Comment:
```suggestion
    Optional<CiviFormProfile> optionalProfile = profileUtils.optionalCurrentUserProfile(requestHeader);
```
(blocking) You can find the [preferred naming of optionals](https://github.com/civiform/civiform/wiki/Development-standards#prefer-optional-instead-of-null) here in the wiki.